base/range.jl

```diff
@@ -116,9 +116,9 @@ _range(a::T, step::T, ::Nothing, len::Integer) where {T <: AbstractFloat} =
 _range(a::T, step, ::Nothing, len::Integer) where {T} =
     _rangestyle(OrderStyle(T), ArithmeticStyle(T), a, step, len)
 _rangestyle(::Ordered, ::ArithmeticWraps, a::T, step::S, len::Integer) where {T,S} =
-    StepRange{T,S}(a, step, convert(T, a+step*(len-1)))
+    StepRange{typeof(a+false*step),S}(a, step, a+step*(len-1))
```

Comment:
`iterate` uses `unsafe_getindex(r, i)` where `i::Int` here, so to be consistent should this be `0*step`?

Reply:
Good point, I will change it.

Reply:
Actually, the `false` in `a+false*step` had a purpose, because this used to work:
```julia
julia> range(typemax(Int)//1, step=1, length=0)
ERROR: OverflowError: 9223372036854775807 + 1 overflowed for type Int64
[...]
```
I will create a new PR that adds the `false` back.